catalogue/src/main/kotlin/com/tidal/sdk/catalogue/Catalogue.kt

```diff
@@ -1,18 +1,59 @@
 package com.tidal.sdk.catalogue
 
-/**
- * Demo documentation block for [Catalogue]
- * @param optInt The constructor [Catalogue] takes an optional [Int]
- * just to tell you about in this comment.
- * As this is used directly by the demo application, which is using
- * Jetpack Compose, this is a data class,which makes it easily comparable
- * and helps prevent unnecessary recomposition
- */
-data class Catalogue(private val optInt: Int = 5) {
+import com.tidal.sdk.auth.CredentialsProvider
+import com.tidal.sdk.catalogue.generated.apis.AlbumJSONAPI
+import com.tidal.sdk.catalogue.generated.apis.ArtistJSONAPI
+import com.tidal.sdk.catalogue.generated.apis.ProviderJSONAPI
+import com.tidal.sdk.catalogue.generated.apis.TrackJSONAPI
+import com.tidal.sdk.catalogue.generated.apis.VideoJSONAPI
+import com.tidal.sdk.catalogue.utils.RetrofitProvider
+
+class Catalogue(credentialsProvider: CredentialsProvider, baseUrl: String = DEFAULT_BASE_URL) {
+
+    private val retrofit = RetrofitProvider().provideRetrofit(baseUrl, credentialsProvider)
```

Comment:
logaf low: this is a rather heavy operation and could be done lazily

Reply:
again, I'll put this on my list for further improvements. As this does not change the API I'll address it separately, just to get this moving.